src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/SNSService.kt

```diff
@@ -1,56 +1,52 @@
-import software.amazon.awssdk.services.sns.SnsClient
-import software.amazon.awssdk.services.sns.model.PublishRequest
+import org.springframework.web.util.UriComponentsBuilder
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.component.SNSPublisher
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.EventDTO
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.ReferralEvent
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.ReferralEventType
 
 @Service
 class SNSService(
-  private val client: SnsClient,
-  private val objectMapper: ObjectMapper,
-  @Value("\${aws.sns.enabled}") private val enabled: Boolean,
-  @Value("\${aws.sns.topic.arn}") private val arn: String,
+  private val snsPublisher: SNSPublisher,
+  @Value("\${interventions-service.baseurl}") private val interventionsServiceBaseURL: String,
+  @Value("\${interventions-service.locations.sent-referral}") private val interventionsServiceSentReferralLocation: String,
 ) : ApplicationListener<ReferralEvent> {
 
   override fun onApplicationEvent(event: ReferralEvent) {
     when (event.type) {
       ReferralEventType.SENT -> {
-        // fixme: what data needs to go in here
-        val message = mapOf(
-          "eventType" to "intervention.referral.sent",
-          "description" to "A referral has been sent to a Service Provider",
-          "referral_id" to event.referral.id,
+        val snsEvent = EventDTO(
+          "intervention.referral.sent",
+          "A referral has been sent to a Service Provider",
+          createDetailUrl(event),
+          event.referral.sentAt!!,
+          1,
+          mapOf("referralId" to event.referral.id)
         )
-        publish(objectMapper.writeValueAsString(message))
+        snsPublisher.publish(snsEvent)
+      }
+      ReferralEventType.ASSIGNED -> {
+        val snsEvent = EventDTO(
+          "intervention.referral.assigned",
+          "A referral has been assigned to a service user",
+          createDetailUrl(event),
+          event.referral.assignedAt!!,
+          1,
+          mapOf("referral_id" to event.referral.id, "assignedTo" to (event.referral.assignedTo?.userName!!))
+        )
+        snsPublisher.publish(snsEvent)
       }
     }
   }
 
-  private fun publish(message: String) {
-    if (!enabled) {
-      return
-    }
-
-    val request = PublishRequest.builder()
-      .message(message)
-      .topicArn(arn)
-      .build()
-
-    try {
-      val result = client.publish(request)
-      log.debug("message sent; messageId: ${result.messageId()}, status: ${result.sdkHttpResponse().statusCode()}")
-    } catch (e: Exception) {
-      log.error("message send failure", e)
-    }
-  }
-
-  companion object {
-    private val log = LoggerFactory.getLogger(SNSService::class.java)
+  private fun createDetailUrl(event: ReferralEvent): String {
+    return UriComponentsBuilder.fromHttpUrl(interventionsServiceBaseURL)
```

Comment:
you can get this from `ServletUriComponentsBuilder.fromCurrentContextPath()` without having to get it from config